packages/ptokens-assets-evm/test/ptokens-evm-provider.test.ts

```diff
@@ -1,530 +1,936 @@
-import PromiEvent from 'promievent'
-import { TransactionReceipt } from 'web3'
+import { TransactionReceipt, createPublicClient, createWalletClient, http } from 'viem'
+import { mainnet, polygon } from 'viem/chains'
 
 import { pTokensEvmProvider } from '../src'
 import * as utils from '../src/lib'
-
 import abi from './utils/exampleContractABI'
 import logs from './utils/logs.json'
+import interimOperationExecutedReceipt from './utils/receiptOperationExecuted.json'
+import factoryAbi from '../src/abi/PFactoryAbi'
+import { poll } from 'viem/utils/poll'
 
 const receiptWithFalseStatus = require('./utils/receiptWithFalseStatus')
 const receiptWithTrueStatus = require('./utils/receiptWithTrueStatus')
 
+const publicClient = createPublicClient({ 
```

Comment:
You should mock these. We can't rely on external factors when testing.
They may change in time and break the tests.